lib/nostrum/struct/guild.ex

```diff
@@ -161,6 +176,55 @@ defmodule Nostrum.Struct.Guild do
   @typedoc since: "0.5.1"
   @type threads :: %{required(Channel.id()) => Channel.t()} | nil
 
+  @typedoc "Hash of the Discovery splash screen"
+  @type discovery_splash :: String.t() | nil
+
+  @typedoc """
+  Bitset representing the system channel flags
+
+  Use `Nostrum.Struct.Guild.SystemChannelFlags` to parse.
+  """
+  @type system_channel_flags :: pos_integer()
+
+  @typedoc "Maximum number of presences for the guild"
+  @type max_presences :: pos_integer() | nil
+
+  @typedoc "Maximum members for the guild"
+  @type max_members :: pos_integer() | nil
+
+  @typedoc "User-set description of the guild"
+  @type description :: String.t() | nil
+
+  @typedoc "Banner hash for the guild, if prefixed with `a_` an animated GIF is available."
+  @type banner :: String.t() | nil
+
+  @typedoc "Premium tier of the guild (0-3)"
+  @type premium_tier :: 0..3
+
+  @typedoc "Number of boosts received by the guild"
+  @type premium_subscription_count :: pos_integer() | nil
+
+  @typedoc "Preferred locale for the guild, set by the user"
+  @type preferred_locale :: String.t() | nil
+
+  @typedoc "The maximum amount of users in a video channel"
+  @type max_video_channel_users :: pos_integer() | nil
+
+  @typedoc "The maximum amount of users in a stage video channel"
+  @type max_stage_video_channel_users :: pos_integer() | nil
```

Comment:
When are these `nil`? Can we document it? Or does `nil` simply mean "no limit"? Should we document that?